packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -920,8 +920,19 @@ export class PipelineEditor extends React.Component<
 
   async handleOpenPipeline(): Promise<void> {
     this.widgetContext.ready.then(async () => {
-      let pipelineJson: any = this.widgetContext.model.toJSON();
-      if (pipelineJson == null) {
+      let pipelineJson: any = null;
+
+      try {
+        pipelineJson = this.widgetContext.model.toJSON();
+      } catch (error) {
+        if (error instanceof SyntaxError) {
```

Comment:
It's ok if we want to have specific handling for SyntaxError, but we should not restrict to only SyntaxError? If there is an error that is not recoverable, don't open the pipeline and let the user know?

Reply:
I agree. 

> don't open the pipeline and let the user know

In that case we open the pipeline editor only to immediately display the dialog to let the user know what's wrong, then clicking OK will close the tab. Is this a valid behaviour?